frontend/components/CheckoutButton/index.jsx

```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import I18n from '@shopgate/pwa-common/components/I18n';
+import Link from '@shopgate/pwa-common/components/Link';
+import RippleButton from '@shopgate/pwa-ui-shared/RippleButton';
+import { CHECKOUT_PATH } from '@shopgate/pwa-common/constants/RoutePaths';
+import connect from './connector';
+import styles from './style';
+
+/**
+ * CheckoutButton renders button
+ * @param {Props} props props.
+ * @returns {JSX}
+ */
+const CheckoutButton = props => (
```

Comment:
Did you maybe consider not implement custom checkout button but wrap the {props.children} with some wrapper that would add some opacity and prevent clicking the original one when checkout is blocked?

We could avoid maintenance of this when PWA6 one would change.

Just a suggesion.

Reply:
Heyo, I didn't think about this before but I did just investigate incorporating context and I think its would be a great way to manipulate isActive with the connector prop isOrderable, but I'm not sure how to incorporate an new onClick function in this way in order to handle the notice mentioned above. I'm sure I must be missing something there.

Reply:
*Was signed into wrong github account so deleted previous comment -_-*
Heyo, I didn't think about this before but I did just investigate incorporating context and I think its would be a great way to manipulate isActive with the connector prop isOrderable, but I'm not sure how to incorporate an new onClick function in this way in order to handle the notice mentioned above. I'm sure I must be missing something there.

Reply:
Please let me know if you have a better idea for this implementation. I pass the original checkout button when orderable.